### ISSUE TITLE
Up-to-date pyload-ng installation guide for Raspberry Pi (OS)?

### ISSUE THREAD

**Issue by fooness** [pyload/pyload#3951]
Hello everyone.

Is there any up-to-date `pyload-ng` installation guide for the Raspberry Pi (OS) for running `pyload` as a service (e.g. via `systemd`)?

```
$ cat /etc/debian_version
10.9
$ uname -a
Linux raspberry 5.10.17-v7l+ #1403 SMP Mon Feb 22 11:33:35 GMT 2021 armv7l GNU/Linux
```

The only guide I found is the following, which was written from 2017 and seems a little outdated. https://github.com/pyload/pyload/wiki/Step-by-Step-Installation-(RaspberryPi)

Thank you very much!

**Comment by GammaC0de** [comment]
Installation is via  `pip install pyload-ng[plugins]`
Then you have to write a systemd service file for it

**Comment by UserX404** [comment]
@GammaC0de
With this approach - how to update to the newest commit/version?  Is pyload-ng self updating or is it handled by `pip install --upgrade pyload-ng` ?

**Comment by GammaC0de** [comment]
@UserX404 Currently a new pypi pyLoad package is automatically generated on every push to the `main` branch.
So `pip install --upgrade pyload-ng` updated pyLoad to the `main` branch.
If you want to keep up with the `develop` branch, you'd have to do it manually.

This is how it is currently, maybe it would change in the future.

**Comment by vuolter** [comment]
@UserX404 remember to append the option `--pre` to the install command to get the latest release (more info in the README) 😉

**Comment by UserX404** [comment]
Thanks for your response. But to be clear: 
pyload-ng: python3, main branch
with --pre: python3, development branch
The old pyload with python2 isn't handled with pip.
Is it correct?

**Comment by vuolter** [comment]
Almost.

I confirm the old pyLoad, working on Python 2, is not available on PyPI.

Regarding `pyload-ng`, as @GammaC0de previously said, we deploy to PyPI on each push occurred to the `main` branch, but the resulting package is marked as `pre-release` on PyPI, so to get it you have to tell pip to fetch also pre-releases with the command option `--pre`, otherwise, you'll get nothing since there's no stable release of pyLoad there right now.
In short, use the command `pip install --pre --upgrade pyload-ng` to always get the latest version.

I also confirm the change applied to the development branch are not deployed on PyPI; if you want to keep synced to our develop branch you can use something like this:
```
pip install --upgrade git+https://github.com/pyload/pyload.git@develop
```
or 
```
pip install --upgrade https://github.com/pyload/pyload/archive/refs/heads/develop.zip
```

**Comment by vuolter** [comment]
@UserX404 let me know if I can close this 😉

**Comment by UserX404** [comment]
@vuolter
You are free to close this. It wasn't my issue but I catched the topic to ask for the difference.
ATM i only ran the command remotly but haven't checked pyload is running. There were some troubles in the past but that's an other story.
Nevertheless thank you for explaination.

**Comment by vuolter** [comment]
oh my, you're right!
Sorry @fooness, waiting for you ofc 😄

**Comment by vuolter** [comment]
Oh my, you're right!
Sorry @fooness, I think there's no updated guide for RPi4 at the moment. 😞

**Comment by fooness** [comment]
Thank you, everyone, for the answers and discussion. So it seems, also for Raspberry Pi 4, `pip` would be the way, right?

Where can I find up-to-date information about creating a `systemd` service file after `pip install …`? And I guess it’s still a good idea to run `pyload` with its own system user?

**Comment by GammaC0de** [comment]
> `pip` would be the way, right?

Right.

> Where can I find up-to-date information about creating a systemd service file after pip install …?

Basically it would be almost the same instructions like the instructions for creating a systemd service in https://github.com/pyload/pyload/wiki/Step-by-Step-Installation-(RaspberryPi) except `ExecStart` would be different.

> And I guess it’s still a good idea to run pyload with its own system user?

It is always wise not to run a service as root if possible.

**Comment by fooness** [comment]
@GammaC0de @vuolter Thank you very much, installation worked. Had to transition my config file from version 1 to version 2, and I’m not, yet, quite sure how to change the external port for the web interface; also I might want to set up HTTPS, again, for which I probably need mkcert … everything else worked flawlessly.

**Comment by fooness** [comment]
One more question, if allowed (because off-topic). The `ExtractArchive` shows only `passwords.txt` in the input field vor “Password file:” … I think I saw the same in the input fields pointing to the ssl cert and key. Where is `passwords.txt` supposed to be located, if not having specified a path of any kind but only the file name. Simply in `.pyload/`?

**Comment by UserX404** [comment]
It doesn't matter. I have just done this 10min ago and created the file within the data directory (for me /opt/var/pyload/data/password.txt).
It may depend if how you would have access to change/extend the file.
At the place i've mentioned i need root/admin permission but you could place the file at every other place with other permission.
The pyload process need to have read access to the file.

**Comment by fooness** [comment]
Seems the passwords.txt file worked! If I could figure out why ssl does not work, and fix it, then I’d be happy!

```
May 05 17:27:05 raspberry pyload[15952]: [2021-05-05 17:27:05]  ERROR               pyload  ADDON ClickNLoad: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1056)
```

```
~/.pyload $ ls
data  logs  passwords.txt  plugins  scripts  settings  ssl.crt  ssl.key
```

The cert and key file were issued via:

```
~/.pyload $ mkcert -cert-file ssl.crt -key-file ssl.key raspberry.local localhost 127.0.0.1 ::1
```

The `webui` snippet from my `pyload.cfg`:

```
webui - "Web Interface":
        bool autologin : "Skip login if single user" = False
        bool develop : "Development mode" = False
        bool enabled : "Activated" = True
        ip host : "IP Address" = localhost
        int port : "Port" = 8000
        str prefix : "Path Prefix" =
        int session_lifetime : "Session lifetime (minutes)" = 44640
        file ssl_certchain : "CA's intermediate certificate bundle (optional)" =
        file ssl_certfile : "SSL Certificate" = ssl.crt
        file ssl_keyfile : "SSL Key" = ssl.key
        Default;modern;pyplex theme : "Theme" = modern
        bool use_ssl : "Use HTTPS" = True
```

**Comment by fooness** [comment]
Maybe I should continue using `pyload` `0.4.20` as supposed here, and just not use `pyload-ng`: https://github.com/pyload/pyload/issues/3762#issuecomment-700929842

I think the README suggests using `pyload-ng`, though?

**Comment by GammaC0de** [comment]
> Maybe I should continue using `pyload 0.4.20` as supposed here, and just not use `pyload-ng`

Why?

**Comment by UserX404** [comment]
I just ran into the same problem.
Is it certificate related? Is there a problem with the c'n'l-Link or the ssl-implementation of pyload?

3002 | 2021-05-05 19:16:04 | DEBUG | pyload | ADDON ClickNLoad: Connection from 192.168.1.2:59046
-- | -- | -- | -- | --
3003 | 2021-05-05 19:16:04 | ERROR | pyload | ADDON ClickNLoad: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1123)
3004 | ? | ? | ? | Traceback (most recent call last):
3005 | ? | ? | ? | File "/opt/lib/python3.9/site-packages/pyload/core/threads/addon_thread.py", line 50, in run
3006 | ? | ? | ? | self.f(*self.args, **self.kwargs)
3007 | ? | ? | ? | TypeError: _server() got an unexpected keyword argument 'thread'
3008 | ? | ? | ? |  
3009 | ? | ? | ? | During handling of the above exception, another exception occurred:
3010 | ? | ? | ? |  
3011 | ? | ? | ? | Traceback (most recent call last):
3012 | ? | ? | ? | File "/opt/lib/python3.9/site-packages/pyload/plugins/addons/ClickNLoad.py", line 198, in _server
3013 | ? | ? | ? | server_socket.connect(self.web_addr)
3014 | ? | ? | ? | File "/opt/lib/python3.9/ssl.py", line 1342, in connect
3015 | ? | ? | ? | File "/opt/lib/python3.9/ssl.py", line 1333, in _real_connect
3016 | ? | ? | ? | File "/opt/lib/python3.9/ssl.py", line 1309, in do_handshake
3017 | ? | ? | ? | ssl.SSLError: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1123)


For me 0.4.20 isn't an option.

**Comment by GammaC0de** [comment]
Does SSL works for the web interface?

**Comment by fooness** [comment]
> > Maybe I should continue using `pyload 0.4.20` as supposed here, and just not use `pyload-ng`
> 
> Why?

Is the following information outdated at the moment?

> You have installed pyload v0.5 which is not ready for use yet.
> Currently the stable version of pyload (v0.4.20) still uses python2.7.

via: https://github.com/pyload/pyload/issues/3762#issuecomment-700929842

**Comment by GammaC0de** [comment]
pyload v0.5 is now beta which means that some of the plugins work and it may have some bugs.

So its up to you to decide.

**Comment by UserX404** [comment]
> 
> 
> Does SSL works for the web interface?

Hmm
I just noticed that i'm not on a ssl connection.
My cert is signed with my public IP but atm i access via local IP.
I expect a warning dialog when switching to https but it seems not to work. But i have ssl activated in the config which seems to be ignored.

webui - "Web Interface":
        bool autologin : "Skip login if single user" = False
        bool develop : "Development mode" = True
        bool enabled : "Activated" = True
        ip host : "IP Address" = 192.168.xx.xx
        int port : "Port" = xxxxx
        str prefix : "Path Prefix" =
        int session_lifetime : "Session lifetime (minutes)" = 44640
        file ssl_certchain : "CA's intermediate certificate bundle (optional)" =
        file ssl_certfile : "SSL Certificate" = /etc/stunnel/backup.cert
        file ssl_keyfile : "SSL Key" = /etc/stunnel/backup.key
        Default;modern;pyplex theme : "Theme" = PyPlex
        bool use_ssl : "Use HTTPS" = True

**Comment by fooness** [comment]
In my case, I cannot access `pyload-ng` externally when SSL is activated. But also I don’t know which port it uses externally; with SSL disabled I can access the webui via port `9666` on which the `clicknload` plugin is listening.

**Comment by UserX404** [comment]
@fooness 
You can check the listening port with netstat.
```
netstat -tulp | grep py
tcp        0      0 0.0.0.0:9666            0.0.0.0:*               LISTEN      2250/python
tcp        0      0 HOSTNAME:PORT         0.0.0.0:*               LISTEN      2250/python
```

In firefox with local and public IP i get an `Errorcode: SSL_ERROR_RX_RECORD_TOO_LONG`
But i have no issue when i access other services of my system via SSL.

@GammaC0de 
Maybe a related topic dealing with urllib3 an a possible solution
https://github.com/urllib3/urllib3/issues/2075#issuecomment-730673871

**Comment by fooness** [comment]
@UserX404 I managed to get `pyload-ng` running on `https://0.0.0.0:8000`, though I cannot access the webui (with the exact same settings in old `pyload` this works) … also, `ClickNLoad` plugin cannot start _when SSL is enabled_ …

```
$ sudo systemctl status pyload.service
* pyload.service - pyload Download Manager
   Loaded: loaded (/etc/systemd/system/pyload.service; enabled; vendor preset: enabled)
   Active: active (running) since Wed 2021-05-05 20:03:31 CEST; 7s ago
 Main PID: 2447 (pyload)
    Tasks: 6 (limit: 4915)
   CGroup: /system.slice/pyload.service
           `-2447 /usr/bin/python3 /usr/local/bin/pyload

May 05 20:03:34 raspberry pyload[2447]: [2021-05-05 20:03:34]  INFO                pyload  Storage directory: /some/folder
May 05 20:03:34 raspberry pyload[2447]: [2021-05-05 20:03:34]  INFO                pyload  Storage free space: XXX.XX GiB
May 05 20:03:34 raspberry pyload[2447]: [2021-05-05 20:03:34]  INFO                pyload  Activating accounts...
May 05 20:03:34 raspberry pyload[2447]: [2021-05-05 20:03:34]  INFO                pyload  Activating Plugins...
May 05 20:03:34 raspberry pyload[2447]: [2021-05-05 20:03:34]  WARNING             pyload  ADDON TORRENT: torrents / magnets are not associated with any plugin
May 05 20:03:34 raspberry pyload[2447]: [2021-05-05 20:03:34]  WARNING       pyload.webui  Starting webserver: https://0.0.0.0:8000
May 05 20:03:39 raspberry pyload[2447]: [2021-05-05 20:03:39]  ERROR               pyload  ADDON ClickNLoad: Could not connect to backend server, ClickNLoad cannot start
```

**Comment by UserX404** [comment]
Keep sure the old installation isn't running.
Check the listening port 9666 and 8000 isn't used by any service
e.g.: `netstat -al | grep 9666`

I just was able to get ssl running by disabling ssl via shell and enable it in UI afterwards.
Strange :-/

@GammaC0de
CnL still throws an exception
```
[2021-05-05 20:34:19]  DEBUG               pyload  ADDON ClickNLoad: Connection from 192.168.1.2:57462
[2021-05-05 20:34:19]  DEBUG               pyload  ADDON ClickNLoad: Connection from 192.168.1.2:57464
Exception in thread Thread-24:
Traceback (most recent call last):
  File "/opt/lib/python3.9/site-packages/pyload/core/threads/addon_thread.py", line 50, in run
    self.f(*self.args, **self.kwargs)
TypeError: forward() got an unexpected keyword argument 'thread'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/lib/python3.9/threading.py", line 954, in _bootstrap_inner
  File "/opt/lib/python3.9/site-packages/pyload/core/threads/addon_thread.py", line 57, in run
    self.f(*self.args, **self.kwargs)
  File "/opt/lib/python3.9/site-packages/pyload/plugins/addons/ClickNLoad.py", line 142, in forward
    forward(source, destination)
  File "/opt/lib/python3.9/site-packages/pyload/plugins/helpers.py", line 564, in forward
    destination.shutdown(socket.SHUT_WR)
  File "/opt/lib/python3.9/ssl.py", line 1280, in shutdown
OSError: [Errno 107] Transport endpoint is not connected
[2021-05-05 20:34:19]  DEBUG               pyload  ADDON ClickNLoad: Connection from 192.168.1.2:57468
Exception in thread Thread-26:
Traceback (most recent call last):
  File "/opt/lib/python3.9/site-packages/pyload/core/threads/addon_thread.py", line 50, in run
    self.f(*self.args, **self.kwargs)
TypeError: forward() got an unexpected keyword argument 'thread'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/lib/python3.9/threading.py", line 954, in _bootstrap_inner
  File "/opt/lib/python3.9/site-packages/pyload/core/threads/addon_thread.py", line 57, in run
    self.f(*self.args, **self.kwargs)
  File "/opt/lib/python3.9/site-packages/pyload/plugins/addons/ClickNLoad.py", line 142, in forward
    forward(source, destination)
  File "/opt/lib/python3.9/site-packages/pyload/plugins/helpers.py", line 564, in forward
    destination.shutdown(socket.SHUT_WR)
  File "/opt/lib/python3.9/ssl.py", line 1280, in shutdown
OSError: [Errno 107] Transport endpoint is not connected
```

**Comment by fooness** [comment]
@UserX404

> Keep sure the old installation isn't running.
> Check the listening port 9666 and 8000 isn't used by any service

Sure … I re-checked, no problems here.

> https://0.0.0.0:8000 is not a valid IP address.

`0.0.0.0` binds to localhost but in a way that is accessible from external devices. At least that’s how it worked it `pyload` `0.4.20` and how it works in e.g. `mpd` and most other services I know.

I can access the webui via http://hostname.local:port with SSL disabled, but not https://hostname.local:port with SSL enabled. Also, ClickNLoad starts with SSL disabled, but not with SSL enabled. The very same setup works in `pyload` `0.4.20`, so the setup seems to be correct.

**Comment by GammaC0de** [comment]
@fooness It seems that the issue happens when you specify 0.0.0.0 as IP, 
For the test, can you try with a valid IP address?

**Comment by fooness** [comment]
@GammaC0de Please excuse the silence; I will be able to answer regularly from Monday on.

> It seems that the issue happens when you specify 0.0.0.0 as IP,
> For the test, can you try with a valid IP address?

I tried with `10.0.0.2`, that’s the static eth0 IP of my raspberry pi … unfortunately, the exact same behavior and error (and error message) as with `0.0.0.0`; connection is not possible, clicknload cannot start.

**Comment by GammaC0de** [comment]
@fooness @UserX404 The issues with ClickNLoad should be fixed, please verify.

**Comment by fooness** [comment]
> @fooness @UserX404 The issues with ClickNLoad should be fixed, please verify.

@GammaC0de I just updated/upgraded `pyload-ng` via the following command:

`pip install --upgrade git+https://github.com/pyload/pyload.git@develop`

Unfortunately, when setting `bool use_ssl : "Use HTTPS" = True`, the problem still exists:

From `~/.pyload/settings/pyload.cfg`:
```
webui - "Web Interface":
        bool autologin : "Skip login if single user" = False
        bool develop : "Development mode" = False
        bool enabled : "Activated" = True
        ip host : "IP address" = 0.0.0.0
        int port : "Port" = 1338
        str prefix : "Path prefix" =
        int session_lifetime : "Session lifetime (minutes)" = 44640
        file ssl_certchain : "CA's intermediate certificate bundle (optional)" =
        file ssl_certfile : "SSL Certificate" = ssl.crt
        file ssl_keyfile : "SSL Key" = ssl.key
        Default;modern;pyplex theme : "Theme" = modern
        bool use_ssl : "Use HTTPS" = True
```

```
$ journalctl -xe -u pyload
Jun 13 13:44:28 raspberry pyload[14298]: [2021-06-13 13:44:28]  INFO               pyload  *** Welcome to pyLoad 0.5.0 ***
Jun 13 13:44:28 raspberry pyload[14298]: [2021-06-13 13:44:28]  DEBUG              pyload  Indexing plugins...
Jun 13 13:44:29 raspberry pyload[14298]: [2021-06-13 13:44:29]  INFO               pyload  Activated plugins: ClickNLoad, ExternalScripts, ExtractArchive, TORRENT, UnSkipOnFail, UserAgentSwitcher
Jun 13 13:44:29 raspberry pyload[14298]: [2021-06-13 13:44:29]  INFO               pyload  Deactivate plugins: AndroidPhoneNotify, AntiStandby, AntiVirus, AppriseNotify, BypassCaptcha, Captcha9Kw, Checksum, ClouJun 13 13:44:30 raspberry pyload[14298]: [2021-06-13 13:44:30]  DEBUG              pyload  Starting core...
Jun 13 13:44:30 raspberry pyload[14298]: [2021-06-13 13:44:30]  DEBUG              pyload  Debug level: TRACE
Jun 13 13:44:30 raspberry pyload[14298]: [2021-06-13 13:44:30]  DEBUG              pyload  Setup language...                                                                                                       Jun 13 13:44:30 raspberry pyload[14298]: [2021-06-13 13:44:30]  DEBUG              pyload  Setup permissions...
Jun 13 13:44:30 raspberry pyload[14298]: [2021-06-13 13:44:30]  INFO               pyload  User directory: /home/user/conf/.pyload                                                                                 Jun 13 13:44:30 raspberry pyload[14298]: [2021-06-13 13:44:30]  INFO               pyload  Cache directory: /tmp/pyLoad                                                                                            Jun 13 13:44:30 raspberry pyload[14298]: [2021-06-13 13:44:30]  INFO               pyload  Storage directory: /mnt
Jun 13 13:44:30 raspberry pyload[14298]: [2021-06-13 13:44:30]  INFO               pyload  Storage free space: XX.XX GiB                                                                                           Jun 13 13:44:30 raspberry pyload[14298]: [2021-06-13 13:44:30]  DEBUG              pyload  Setup network...
Jun 13 13:44:30 raspberry pyload[14298]: [2021-06-13 13:44:30]  INFO               pyload  Activating accounts...
[...]
Jun 13 13:44:32 raspberry pyload[14298]: [2021-06-13 13:44:32]  INFO               pyload  Activating Plugins...
Jun 13 13:44:32 raspberry pyload[14298]: [2021-06-13 13:44:32]  DEBUG              pyload  ADDON ExtractArchive: Found HjSplit 0.02 | Found UnRar 5.61 | Found UnZip 3.7.3 | Found UnTar 3.7.3
Jun 13 13:44:32 raspberry pyload[14298]: [2021-06-13 13:44:32]  DEBUG              pyload  ADDON ExternalScripts: No script found under folder `pyload_start`
Jun 13 13:44:32 raspberry pyload[14298]: [2021-06-13 13:44:32]  WARNING            pyload  ADDON TORRENT: torrents / magnets are not associated with any plugin
Jun 13 13:44:32 raspberry pyload[14298]: [2021-06-13 13:44:32]  WARNING      pyload.webui  Starting webserver: https://0.0.0.0:1338
Jun 13 13:44:32 raspberry pyload[14298]: [2021-06-13 13:44:32]  DEBUG              pyload  *** pyLoad is up and running ***
Jun 13 13:44:35 raspberry pyload[14298]: [2021-06-13 13:44:35]  ERROR              pyload  ADDON ClickNLoad: Could not connect to backend server, ClickNLoad cannot start
```

**Without** SSL, it looks like this:
```
$ journalctl -xe -u pyload
[...]
Jun 13 13:54:59 raspberry pyload[14434]: [2021-06-13 13:54:59]  WARNING      pyload.webui  Starting webserver: http://0.0.0.0:1338
Jun 13 13:54:59 raspberry pyload[14434]: [2021-06-13 13:54:59]  DEBUG              pyload  *** pyLoad is up and running ***
Jun 13 13:55:02 raspberry pyload[14434]: [2021-06-13 13:55:02]  DEBUG              pyload  ADDON ClickNLoad: Backend found on http://127.0.0.1:1338
Jun 13 13:55:02 raspberry pyload[14434]: [2021-06-13 13:55:02]  INFO               pyload  ADDON ClickNLoad: Proxy listening on 0.0.0.0:9666
```

Any idea what else I could try, or what more information I could provide?

**Comment by fooness** [comment]
PS: Confirming it’s fixed.

I seems that indeed the paths to the `ssl.crt` and `ssl.key` file needs to be either absolute or relative **to the home directory** (this was most probably not the case in old pyload, but it was relative to the `~/.pyload` directory).

```
webui - "Web Interface":
        bool autologin : "Skip login if single user" = False
        bool develop : "Development mode" = False
        bool enabled : "Activated" = True
        ip host : "IP address" = 0.0.0.0
        int port : "Port" = 1338
        str prefix : "Path prefix" =
        int session_lifetime : "Session lifetime (minutes)" = 44640
        file ssl_certchain : "CA's intermediate certificate bundle (optional)" =
        file ssl_certfile : "SSL Certificate" = .pyload/ssl.crt
        file ssl_keyfile : "SSL Key" = .pyload/ssl.key
        Default;modern;pyplex theme : "Theme" = modern
        bool use_ssl : "Use HTTPS" = True
```

Related to that message above: https://github.com/pyload/pyload/issues/3951#issuecomment-832096420

> One more question, if allowed (because off-topic). The `ExtractArchive` shows only `passwords.txt` in the input field for “Password file:” … I think I saw the same in the input fields pointing to the ssl cert and key. Where is `passwords.txt` supposed to be located, if not having specified a path of any kind but only the file name. Simply in `.pyload/`?

Let me check if the old(?) behavior could be achieved via `str prefix : "Path prefix" = .pyload/` …

**Comment by GammaC0de** [comment]
> PS: Confirming it’s fixed.

Great, @UserX404 Is it working for you too?

>> One more question, if allowed (because off-topic). The ExtractArchive shows only passwords.txt in the input field for “Password file:” … I think I saw the same in the input fields pointing to the ssl cert and key. Where is passwords.txt supposed to be located, if not having specified a path of any kind but only the file name. Simply in .pyload/?

Actually it was relative to the user home directory, now changed to the userdir/data.

**Comment by fooness** [comment]
> > > One more question, if allowed (because off-topic). The ExtractArchive shows only passwords.txt in the input field for “Password file:” … I think I saw the same in the input fields pointing to the ssl cert and key. Where is passwords.txt supposed to be located, if not having specified a path of any kind but only the file name. Simply in .pyload/?
> 
> Actually it was relative to the user home directory, now changed to the userdir/data.

So I could move `ssl.crt`, `ssl.key`, and `passwords.txt` to `~/.pyload/data/` and then specifying the filenames without a path in the webui would work? Will try to confirm that in the coming days.

**Comment by GammaC0de** [comment]
That is the plan.

**Comment by GammaC0de** [comment]
@UserX404 Can we close this issue?

**Comment by kjboa** [comment]
From what i've read here so far and what i've found on google it is not really clear to me how to define pyload as a system service. Could someone explain to me?
